client/lib/site/jetpack.js

```diff
@@ -249,63 +214,53 @@ JetpackSite.prototype.callHome = function() {
 };
 
 JetpackSite.prototype.activateModule = function( moduleId, callback ) {
-	var module = this.getModule( moduleId );
 	debug( 'activate module', moduleId );
 
-	if ( ! module.id ) {
+	if ( ! moduleId ) {
 		callback && callback( new Error( 'No id' ) );
 		return;
 	}
 
-	if ( module.active ) {
+	if ( this.isModuleActive( moduleId ) ) {
 		// Nothing to do
 		callback();
 		return;
 	}
 
-	this.toggleModule( module, callback );
+	this.toggleModule( moduleId, callback );
 };
 
 JetpackSite.prototype.deactivateModule = function( moduleId, callback ) {
-	var module = this.getModule( moduleId );
 	debug( 'deactivate module', moduleId );
 
-	if ( ! module.id ) {
-		callback && callback( new Error( 'No id' ) );
-		return;
-	}
-
-	if ( ! module.active ) {
+	if ( ! this.isModuleActive( moduleId ) ) {
 		// Nothing to do
 		callback();
 		return;
 	}
 
-	this.toggleModule( module, callback );
+	this.toggleModule( moduleId, callback );
 };
 
-JetpackSite.prototype.toggleModule = function( module, callback ) {
-	var method;
+JetpackSite.prototype.toggleModule = function( moduleId, callback ) {
+	const isActive = this.isModuleActive( moduleId ),
+	 	method = isActive ? 'jetpackModulesDeactivate' : 'jetpackModulesActivate',
+		prevActiveModules = Object.assign( {}, this.modules );
```

Comment:
could we name this `prevStateModules` or something like that? if I'm not understanding it wrong, it saves the previous state of the modules to be able to restore them if there's an error, but the name suggest me a subset of the modules that includes only the active ones